lib/cylc/scheduler.py

```diff
@@ -388,9 +388,9 @@ def command_set_stop_cleanly(self, kill_active_tasks=False):
         """Stop job submission and set the flag for clean shutdown."""
         SuiteProcPool.get_inst().stop_job_submission()
         TaskProxy.stop_sim_mode_job_submission = True
-        if kill_active_tasks:
-            self.pool.kill_active_tasks()
         self.shut_down_cleanly = True
+        self.kill_on_shutdown = kill_active_tasks
+        self.next_kill_issue = datetime.datetime.utcnow()
```

Comment:
(A gist to prove the claim above: https://gist.github.com/matthewrmshin/cabf70970b2dc7e85123)